core/modules/hubot/robot.js

```diff
@@ -0,0 +1,223 @@
+﻿var Message = require.once('./message.js'),
+    Responder = require.once('./responder.js'),
+    fs = require('fs'),
+    path = require('path');
+
+var Robot = function(Instance, descriptor, config) {
+    this.name = descriptor.name;
+    this._help = descriptor.help;
+    this.folderPath = descriptor.folderPath;
+
+    this.listeners = [];
+    this.catchAllListeners = [];
+    this.instances = [];
+
+    this.brain = {
+        on: function(event, callback) {
+            if (event === 'loaded') {
+                callback();
+            }
+        },
+        emit: function() {},
+        data: config
+    };
+
+    this.instances.push(new Instance(this));
+};
+
+Robot.generateHubotJson = function (folderPath, scriptLocation) {
+    var hubotDocumentationSections = [
+            'description',
+            'dependencies',
+            'configuration',
+            'commands',
+            'notes',
+            'author',
+            'authors',
+            'examples',
+            'tags',
+            'urls'
+        ],
+        mod = path.join(folderPath, scriptLocation),
+        body = fs.readFileSync(mod, 'utf-8'),
+        scriptDocumentation = { name: path.basename(mod).replace(/\.(coffee|js)$/, '') },
+        currentSection = null,
+        ref = body.split("\n"),
+        commands = [];
+
+    for (var i = 0; i < ref.length; i++) {
+        var line = ref[i];
+        if (line[0] !== '#' && line.substr(0, 2) !== '//') {
+            break;
+        }
+        var cleanedLine = line.replace(/^(#|\/\/)\s?/, "").trim();
+        if (cleanedLine.length === 0 || cleanedLine.toLowerCase() === 'none') {
+            continue;
+        }
+
+        var nextSection = cleanedLine.toLowerCase().replace(':', '');
+        if (hubotDocumentationSections.indexOf(nextSection) >= 0) {
+            currentSection = nextSection;
+            scriptDocumentation[currentSection] = [];
+        } else if (currentSection) {
+            scriptDocumentation[currentSection].push(cleanedLine.trim());
+            if (currentSection === 'commands') {
+                commands.push(cleanedLine.trim());
+            }
+        }
+    }
+
+    var help = [];
+    if (scriptDocumentation.commands) {
+        for (var i = 0; i < scriptDocumentation.commands.length; i++) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'i' is already declared in line 48.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530104138/issues/source?bid=3282987&fileBranchId=3282997#l72)